code/modules/mining/equipment/lazarus_injector.dm

```diff
@@ -11,52 +18,53 @@
 	w_class = WEIGHT_CLASS_SMALL
 	throw_speed = 3
 	throw_range = 5
-	var/loaded = 1
-	var/malfunctioning = 0
-	var/revive_type = SENTIENCE_ORGANIC //So you can't revive boss monsters or robots with it
+	///Can this still be used?
+	var/loaded = TRUE
+	///Injector malf?
+	var/malfunctioning = FALSE
+	///So you can't revive boss monsters or robots with it
+	var/revive_type = SENTIENCE_ORGANIC
 
 /obj/item/lazarus_injector/afterattack(atom/target, mob/user, proximity_flag)
 	. = ..()
-	if(!loaded)
+	if(!loaded || !(isliving(target) && proximity_flag) )
+		return
+	if(!isanimal(target))
+		to_chat(user, span_info("[src] is only effective on lesser beings."))
+		return
+
+	var/mob/living/simple_animal/M = target
+	if(M.sentience_type != revive_type)
+		to_chat(user, span_info("[src] does not work on this sort of creature."))
+		return
+	if(M.stat != DEAD)
+		to_chat(user, span_info("[src] is only effective on the dead."))
 		return
-	if(isliving(target) && proximity_flag)
-		if(isanimal(target))
-			var/mob/living/simple_animal/M = target
-			if(M.sentience_type != revive_type)
-				to_chat(user, span_info("[src] does not work on this sort of creature."))
-				return
-			if(M.stat == DEAD)
-				M.faction = list("neutral")
-				M.revive(full_heal = TRUE, admin_revive = TRUE)
-				if(ishostile(target))
-					var/mob/living/simple_animal/hostile/H = M
-					if(malfunctioning)
-						H.faction |= list("lazarus", "[REF(user)]")
-						H.robust_searching = 1
-						H.friends += user
-						H.attack_same = 1
-						log_game("[key_name(user)] has revived hostile mob [key_name(target)] with a malfunctioning lazarus injector")
-					else
-						H.attack_same = 0
-				loaded = 0
-				user.visible_message(span_notice("[user] injects [M] with [src], reviving it."))
-				SSblackbox.record_feedback("tally", "lazarus_injector", 1, M.type)
-				playsound(src,'sound/effects/refill.ogg',50,TRUE)
-				icon_state = "lazarus_empty"
-				return
-			else
-				to_chat(user, span_info("[src] is only effective on the dead."))
-				return
+
+	M.faction = list("neutral")
+	M.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(ishostile(target))
+		var/mob/living/simple_animal/hostile/H = M
```

Comment:
Can you give these `M` and `H` variables better names?

Reply:
done